guides/doc-Provisioning_Guide/topics/proc_configuring_iPXE_to_reduce_provisioning_times.adoc

```diff
@@ -69,32 +50,76 @@ Before you begin, ensure that the following conditions are met:
 For more information, see https://ipxe.org/appnote/hardware_drivers[supported hardware on ipxe.org] for a list of hardware drivers expected to work with an iPXE-based boot disk.
 * The NIC is compatible with iPXE.
 
-[[setting_up_the_discovery_service_for_iPXE]]
-=== Setting up the Discovery Service for iPXE
+.Procedure
 
-. On {SmartProxyServer}, install the Foreman discovery service:
+. To prepare iPXE environment, you must perform this procedure on all {SmartProxies}.
++
+. Enable the *tftp* and *httpboot* services:
 +
 [options="nowrap" subs="+quotes,attributes"]
-ifdef::satellite[]
 ----
-# {package-install} foreman-discovery-image
+# {foreman-installer} --foreman-proxy-httpboot true --foreman-proxy-tftp true
 ----
-endif::[]
-
-. On {SmartProxyServer}, enable the *httpboot* service:
++
+. Install the `ipxe-bootimgs` RPM package:
 +
 [options="nowrap" subs="+quotes,attributes"]
 ----
-# {foreman-installer} --foreman-proxy-httpboot true
+# {package-install} ipxe-bootimgs
+----
++
+ifdef::foreman-el,katello[]
+. On Debian/Ubuntu, install the `ipxe` .deb package:
++
+----
+# {package-install} ipxe
+----
++
+endif::[]
+ifdef::foreman-el,katello[]
+. On systems with SELinux, correct the file contexts:
++
+----
+# restorecon -RvF /var/lib/tftpboot/
+----
+endif::[]
+ifdef::satellite,orcharhino[]
+. Correct the SELinux file contexts:
++
+----
+# restorecon -RvF /var/lib/tftpboot/
+----
+endif::[]
++
+* Copy the iPXE firmware with Linux kernel header to the TFTP directory:
++
+----
+# cp /usr/share/ipxe/ipxe.lkrn /var/lib/tftpboot/
+----
++
+* Copy the UNDI iPXE firmware to the TFTP directory:
++
 ----
+# cp /usr/share/ipxe/undionly.kpxe /var/lib/tftpboot/undionly.0
+----
++
+. Optionally, configure Foreman discovery:
++
+* Install Foreman discovery image:
 +
-. In the {ProjectWebUI}, navigate to *Administer* > *Settings*, and click the *Provisioning* tab.
-. Locate the *Default PXE global template entry* row and in the *Value* column, change the value to *discovery*.
+[options="nowrap" subs="+quotes,attributes"]
+ifdef::satellite[]
```

Comment:
Is this ifdef=satellite really necessary? I think this also applies to Foreman and orcharhino.

Reply:
There is no RPM package upstream, let me do a xref instead.